source/MaterialXGenMdl/MdlShaderGenerator.cpp

```diff
@@ -208,6 +208,11 @@ MdlShaderGenerator::MdlShaderGenerator() :
 
 ShaderPtr MdlShaderGenerator::generate(const string& name, ElementPtr element, GenContext& context) const
 {
+    // For MDL we cannot cache node implementations between generation calls,
```

Comment:
Don't recall if there a virtual method to "initialize" before each generation? Maybe it should go there instead or add one.
Just a suggestion so it's obvious.